app/src/main/java/com/duckduckgo/app/statistics/model/PixelEntity.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2020 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.statistics.model
+
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+import androidx.room.TypeConverter
+import com.duckduckgo.app.di.JsonModule
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Types
+
+@Entity(tableName = "pixel_store")
+data class PixelEntity(
+    @PrimaryKey(autoGenerate = true) var id: Long = 0,
+    val pixelName: String,
+    val atb: String,
+    val additionalQueryParams: Map<String, String> = emptyMap(),
+    val encodedQueryParams: Map<String, String> = emptyMap()
+)
+
+class QueryParamsTypeConverter {
+    @TypeConverter
+    fun toQueryParams(value: String): Map<String, String> {
+        return Adapters.queryParamsAdapter.fromJson(value)!!
+    }
+
+    @TypeConverter
+    fun fromQueryParams(value: Map<String, String>): String {
+        return Adapters.queryParamsAdapter.toJson(value)
+    }
+}
+
+class Adapters {
+    companion object {
+        private val moshi = JsonModule().moshi()
```

Comment:
It's a bit strange having to instantiate a dagger module, to then instantiate a `Moshi` instance which is marked as `Singleton`.

I don't know how big of a deal that is, though, but feels _off_ to me. Any thoughts on if it's a problem, or if there's a nicer way to handle this?

Reply:
This is not really an issue, when I first saw it, it also felt odd, but in the end this is bypassing all DI and just creating an instance of moshi. Only that is creating one with the same configuration that we use in the instances provided by DI.
Imo it would suffice to just do `Moshi.Builder().build()`. Prob we don't even need to add any other adapter for this.

Note: this pattern is also done in `TdsTracker.ky`

Reply:
Yeah, it's a pattern we use in different parts. I don't think is an issue neither, just an implementation detail on how to go from Map to String in order to store it in the database. I agree that could be confusing because we use `Moshi` also to communicate with different API, but it's unrelated to that instance we use there. Also better to keep away other adapters that could add unexpected transformations.

Reply:
We could consider pulling out the builder (of the builder) somewhere outside of the Dagger modules at some point, so that both dagger and anything else that needs it go there, instead of everything going to Dagger module. Not a big deal; just odd.

Happy to leave as is, though 👍